oneview_redfish_toolkit/api/zone.py

```diff
@@ -81,18 +80,42 @@ def __init__(self, profile_template, available_targets_obj, drives=[]):
 
         self._validate()
 
-    def fill_resource_blocks(self, profile_template, available_targets,
+    def fill_resource_blocks(self, profile_template, enclosure_group_resource,
                              drives):
-        for item in available_targets:
-            if item["serverHardwareUri"]:
-                self.add_resource_block_item_to_links(item,
-                                                      "serverHardwareUri")
+        server_hardware_list = \
+            self._get_server_hardware_list(enclosure_group_resource)
+
+        for item in server_hardware_list:
+            self.add_resource_block_item_to_links(item, "uri")
 
         for item in drives:
             self.add_resource_block_item_to_links(item, "uri")
 
         self._fill_network_resource_block(profile_template)
 
+    def _get_server_hardware_list(self, encl_group):
+        server_hardware_list = list()
+        logical_encl = \
+            encl_group["children"]["ENCLOSURE_GROUP_TO_LOGICAL_ENCLOSURE"][0]
+        enclosures = logical_encl["children"]["LOGICAL_ENCLOSURE_TO_ENCLOSURE"]
```

Comment:
Yes, I agree you. But, there is other way to get the server hardware list that it will simplify our life. =)
We can get server hardware by enclosure group directly: `https://{{oneview-ip}}/rest/server-hardware?filter="'serverGroupUri'='/rest/enclosure-groups/bc41f38d-e8ce-4241-acd1-00b2d8c5d0fa'"` Or using the python sdk: `oneview_client.server_hardware.get_all(filter='''serverGroupUri'='/rest/enclosure-groups/bc41f38d-e8ce-4241-acd1-00b2d8c5d0fa'")`

I believe @karolcatunda  already started doing this change.